File: sample/sample/AdOverlayPreloadViewController.m
//
//  AdOvrelayPreloadViewController.m
//  sample
//
//  Created by Vladas Drejeris on 04/05/15.
//  Copyright (c) 2015 Adform. All rights reserved.
//

#import "AdOverlayPreloadViewController.h"

@interface AdOverlayPreloadViewController () <AFAdOverlayDelegate>

@property (nonatomic, weak) IBOutlet UIActivityIndicatorView *indicatorView;
@property (nonatomic, weak) IBOutlet UIButton *showButton;

@end

@implementation AdOverlayPreloadViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    
    self.adOverlay.delegate = self;
    
    [self.indicatorView startAnimating];
    
    // Initiate overlay ad loading
    [self.adOverlay preloadAd];
}

- (IBAction)showAd:(id)sender {
    
    [self.adOverlay showFromViewController:self];
    
    self.showButton.hidden = YES;
}

#pragma mark - AFAdOverlayDelegate

- (void)adOverlayDidLoadAd:(AFAdOverlay *)adOverlay {
    
    [self.indicatorView stopAnimating];
    self.showButton.hidden = NO;
}

@end

File: sample/sample/Placements/AdOverlayPreloadViewController.swift
//
//  AdOverlayPreloadViewController.swift
//  sample
//
//  Created by Laurynas Letkauskas on 2021-02-26.
//

import UIKit
import AdformAdvertising

class AdOverlayPreloadViewController: AdOverlayViewController {
    @IBOutlet private weak var indicatorView: UIActivityIndicatorView!
    @IBOutlet private weak var showButton: UIButton!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view.

        adOverlay?.delegate = self

        indicatorView.startAnimating()

        // Initiate overlay ad loading
        adOverlay?.preloadAd()
    }

    @IBAction func showAd(_ sender: Any) {

        adOverlay?.show(from: self)

        showButton.isHidden = true
    }
}

extension AdOverlayPreloadViewController: AFAdOverlayDelegate {
    
    func adOverlayDidLoadAd(_ adOverlay: AFAdOverlay) {
        indicatorView.stopAnimating()
        showButton.isHidden = false
    }
}
